lib/ensure_content_type.js

```diff
@@ -72,8 +72,9 @@ function checkContentType(hyper, req, next, expectedContentType, responsePromise
                     expected: expectedContentType,
                     actual: res.headers['content-type']
                 });
-                // Limit response caching to a few seconds
-                res.headers['cache-control'] = 'max-age=10, s-maxage=10';
+                // Disable response caching, as we aren't setting a vary
+                // either.
+                res.headers['cache-control'] = 'max-age=0, s-maxage=0';
```

Comment:
Overriding caching in a filter is always a bit tricky, since we might actually end up unintentionally *extending* the cache period in some cases. Avoiding that would require parsing existing cache-control headers, which adds complexity. Given that such upgrade failures are error conditions, should be very rare, and we can't set a very long TTL anyway, it does not seem to be worth the complexity or risk to me.